addons/account/models/account.py

```diff
@@ -226,20 +225,53 @@ def load(self, fields, data):
         return rslt
 
     @api.multi
+    def _toggle_reconcile_to_true(self):
+        query = """
+            UPDATE account_move_line
+                SET reconciled = CASE
+                    WHEN debit = 0 AND credit = 0 AND amount_currency = 0
+                    THEN true
+                    ELSE false
+                    END,
+                full_reconcile_id = NULL,
```

Comment:
I find this update of the `full_reconcile_id` a little bit dangerous for the user.
Case :
1- the user set to false the reconcile flag of the account
2- the user say : I have made an mistake
3- the user set to true
4- the user said "why all the matching number have been lost"

NOTA : I don't understand to set `reconciled = True` for a line with `debit = 0 and credit = 0`

Reply:
@qdp-odoo what this think about to set NULL on full_reconcile_id ?

Reply:
@fmdl ok, I propose to add the clause 'and full_reconcile_id is null' in the query to make sure we don't alter already fully reconciled entries
cc @smetl